### PR TITLE
ENG-38732: adding metrics config

### PR DESCRIPTION
**Pull request by varkey98**
Adding Metrics config. As of now only sampling metrics is captured and they're just being print in the logs

### CODE REVIEW

File: proto/ai/traceable/agent/config/v1/config.proto [hunk 1]
```diff
@@ -16,6 +16,7 @@ message AgentConfig {
   // javaagent has the configs specific to javaagent
   Javaagent javaagent = 7;
   LogConfig logging = 8;
+  MetricsConfig metrics_config = 9;
```
Comment (shashank11p) [inline proto/ai/traceable/agent/config/v1/config.proto:19]: Which metrics will be covered in this metrics_config?

File: proto/ai/traceable/agent/config/v1/config.proto [hunk 2]
```diff
@@ -130,3 +131,21 @@ message LogFileConfig {
...
   google.protobuf.StringValue file_path = 3;
 }
+
+message EndpointMetricsConfig {
+  // set this flag to enable endpoint level metrics
+  google.protobuf.BoolValue enabled = 1;
+  // set the frequency at which endpoint level metrics should be printed. Examples are '1s', '2m', '3h'. Default value is 30m
+  google.protobuf.StringValue frequency = 2;
+  // set the max number of endpoints to track
+  google.protobuf.Int32Value max_endpoints = 3;
```
[the remark this answers is not in the document]
Comment (varkey98) [reply proto/ai/traceable/agent/config/v1/config.proto:141]: Yes

File: proto/ai/traceable/agent/config/v1/config.proto [hunk 3]
```diff
@@ -130,3 +131,21 @@ message LogFileConfig {
   // file path for the log file. Default value is /var/traceable/log/libtraceable.log
   google.protobuf.StringValue file_path = 3;
 }
+
+message EndpointMetricsConfig {
+  // set this flag to enable endpoint level metrics
+  google.protobuf.BoolValue enabled = 1;
+  // set the frequency at which endpoint level metrics should be printed. Examples are '1s', '2m', '3h'. Default value is 30m
```
[the remark this answers is not in the document]
Comment (varkey98) [reply proto/ai/traceable/agent/config/v1/config.proto:138]: right now we just print the stats in the log

File: proto/ai/traceable/agent/config/v1/config.proto (hunk 1 above)
Comment (varkey98) [reply proto/ai/traceable/agent/config/v1/config.proto:19]: For now, we just have sampling metrics

**Review by shashank11p — APPROVED** [review]

**Pull request merged**